### PR TITLE
initial description of UIViewController and related things

### PR DESCRIPTION
**Pull request by r10s**
the idea here is to have a document that:
- defines class names and concepts really _needed_
- some of the _conventions_
- and finally describes what we are actually doing and why this is sufficient for Delta Chat (eg. extra-separation of Data an View may not be needed as it is already separated as all data are handled by the core)

related questions: https://github.com/deltachat/deltachat-ios/issues/295, https://github.com/deltachat/deltachat-ios/issues/323

@cyBerta @nayooti first of all, can you please have a look at the _definitions_ section and see if this is correct or if sth. important is missing.

### CODE REVIEW

File: docs/uiview-layout.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  - the functions in the "delegate" are called by the "delegator"
+  - both, "delegator" and "delegate" may be view controllers,
+    eg. ViewControllerA ("delegator") is sending an event to ViewControllerB ("delegate")
+  - in practise, subclasses just override functions defined by "delegate" protocol, as `didSelect()`
+  - used view controllers the system provides
+
+- "coordinator"
+  - convention, no requirement from UIKit
+  - defined as an empty protocol class from where other derive
+  - they take the view controller as argument on construction
```
Comment (nayooti) [inline docs/uiview-layout.md:36]: they take an **UINavigationController** as argument.

File: docs/uiview-layout.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    and call eg. `pushViewController()` to go to other views
```
Comment (nayooti) [inline docs/uiview-layout.md:37]: ... to navigate to other viewControllers.

File: docs/uiview-layout.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  - a "delegate" (see "view delegate" for definition)
+    for events on application-level as `applicationWillEnterForeground()`
+  - we need to derive from `UIApplicationDelegate` and pass this to `UIApplicationMain`
+  - `UIApplication.shared.delegate` will keep a reference to our delegate
+  - we can store "app globals" here, as DcContext
+
+- `UIWindow`
+  - backdrop for the ui, dispatches events, derived from `UIView`
+  - stored in `UIApplication.shared.delegate.window`,
+  - created in `application(_:didFinishLaunchingWithOptions:)` - TODO: is that correct?
```
Comment (nayooti) [inline docs/uiview-layout.md:68]: In pure iOS13 apps `window` is actually set in sceneDelegate.

**Comment by r10s** [comment]
i merge that for now, the plan is to update the section "what is needed? what adds unneeded complexity?" together with the upcoming improvements

**Pull request merged**